### DISCUSSION TITLE
.get() only available for id and not for other parameters?

### DISCUSSION THREAD

**Discussion by jayGG83** [netbox-community/netbox-python#17] [Q&A]
Hello,
As mentioned, by default any get function only accepts the object id as parameter. 
Is there any onther way, like getting the info by name?

Thanks

**Comment by markkuleinio** [comment]
`.get(name="xxx")`?

**Reply by jayGG83** [reply]
Already tried that and it doesn't work. I get:
```
ERROR:root:No device found with name TEST_1: RetrievableAPIResource.get() got an unexpected keyword argument 'name'
```

Checking the code, get has 2 classes: 
```
class RetrievableAPIResource:
    def get(self, id: str | int) -> Result:
        return self.client.get(f"{self.path}{id}/")


class RetrievableRootAPIResource:
    def get(self, **kwargs) -> Result:
        return self.client.get(self.path, params=kwargs)
```
The seond one is the one that enables the search throuygh other parameters, but only one used is only the first one:
```
class APIResource(
    baseapi,
    CreateableAPIResource,
    DeletableAPIResource,
    ListableAPIResource,
    RetrievableAPIResource,
    UpdateableAPIResource,
):
    pass
```

**Reply by markkuleinio** [reply]
Sorry, I thought this was in pynetbox repo and not in netbox-python.

**Reply by jayGG83** [reply]
I'm trying to get module information, but Pynetbox doesn't have it covered. netbox-python has it, but the .get is only enabled for ids :(
